src/mscorlib/src/System/ThrowHelper.cs

```diff
@@ -53,6 +53,12 @@ internal static class ThrowHelper {
             throw new ArgumentException(Environment.GetResourceString("Arg_WrongType", value, targetType), "value");
         }
 
+#if FEATURE_CORECLR
+        internal static void ThrowAddingDuplicateWithKeyArgumentException(object key) {
+            throw new ArgumentException(Environment.GetResourceString("Argument_AddingDuplicateWithKey", key));
```

Comment:
That's evil :laughing: 

See also @nguerrera's comment in https://github.com/dotnet/corefx/pull/185#issuecomment-65754562